src/server.lisp

```diff
@@ -297,9 +303,17 @@ Argument descriptions:
   (check-type debug boolean)
   (check-type listen-addresses list)
   (let ((pool-address (format nil "inproc://~a" (uuid:make-v4-uuid))))
-    (cl-syslog:format-log logger ':info
-                          "Spawning server at ~a .~%" listen-addresses)
-    (pzmq:with-sockets ((clients :router :monitor) (workers :dealer :monitor))
+    (cl-syslog:format-log logger ':info "Spawning server at ~a .~%" listen-addresses)
+    (pzmq:with-sockets ((clients :router) (workers :dealer))
+      ;; Configuring the server secret key here enables encryption on the socket and allows clients
+      ;; to authenticate the server. However, this server is not currently authenticating connected
+      ;; client keys. In order to do, we'd need to implement that authentication ourselves on top of
+      ;; the ZeroMQ Authentication Protocol (ZAP) (unless such support is added to PZMQ).
+      ;;
+      ;; https://rfc.zeromq.org/spec/27/
```

Comment:
Adding an authenticator for client keys shouldn't be hard, but it does add some complication to an already complicated server loop, so I opted to leave it out until we actually need it.

Reply:
Also, fwiw not authenticating clients corresponds to the first security model described in the [CURVEZMQ spec](https://rfc.zeromq.org/spec/26/#general-design):

 > In a CurveZMQ architecture, the clients MUST know the server public key before they can connect. Servers MAY know clients’ public keys, and MAY distinguish different clients based on their keys. This gives us three possible security models:
>
> - Where the server does not check client keys at all. In this case the clients can be certain they are talking securely to the correct server, but the server will accept connections from any client. This fits the conventional Internet model where a browser talks securely to a website to place and order and send credit card information.